### PR TITLE
Fix bug in create_meshgrid3d along depth

### PR DESCRIPTION
**Pull request by shiyangc-intusurg**
#### Changes
fix a typo due to copypaste.
otherwise, size and value along the depth axis could be wrong when normalized_coordinates = True
No dependencies are required for this change.

Fixes # (issue)

#### Type of change
- [x] 🐞 Bug fix (non-breaking change which fixes an issue)

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

**Review by edgarriba — APPROVED** [review]

File: kornia/utils/grid.py [hunk 1]
```diff
@@ -95,7 +95,7 @@ def create_meshgrid3d(
     if normalized_coordinates:
         xs = (xs / (width - 1) - 0.5) * 2
         ys = (ys / (height - 1) - 0.5) * 2
-        zs = (ys / (height - 1) - 0.5) * 2
+        zs = (zs / (depth - 1) - 0.5) * 2
     # generate grid by stacking coordinates
     base_grid: torch.Tensor = torch.stack(torch.meshgrid([zs, xs, ys])).transpose(1, 2)  # 3xHxW
     return base_grid.unsqueeze(0).permute(0, 3, 4, 2, 1)  # 1xHxWx3
```
Comment (edgarriba) [inline kornia/utils/grid.py:101]:
```suggestion
    return base_grid.unsqueeze(0).permute(0, 3, 4, 2, 1)  # 1xDxHxWx3
```

Comment (edgarriba) [reply kornia/utils/grid.py:101]: I think also that the transpose/permute could be done just once

Comment (edgarriba) [reply kornia/utils/grid.py:101]: Also seeing that we don’t have proper tests here. Otherwise this change might have flagged up some errors /cc @shijianjian

**Review by edgarriba — COMMENTED** [review]
Also this line should be modified: https://github.com/kornia/kornia/blob/master/kornia/augmentation/augmentation3d.py#L944

plus, @shijianjian `RandomPerspective3D` don’t appear in the docs

**Comment by shiyangc-intusurg** [comment]
@edgarriba I added another comment to merge transpose and permute.
However, I don't think what you mentioned (here)[https://github.com/kornia/kornia/pull/1330#pullrequestreview-765021230] is related to this pull request.

File: kornia/utils/grid.py (hunk 1 above)
Comment (shiyangc-intusurg) [reply kornia/utils/grid.py]: @edgarriba I added another comment to merge transpose and permute.
However, I don't think what you mentioned here is related to this pull request.

**Comment by shiyangc-intusurg** [comment]
> Also this line should be modified: https://github.com/kornia/kornia/blob/master/kornia/augmentation/augmentation3d.py#L944
> 
> plus, @shijianjian `RandomPerspective3D` don’t appear in the docs



> Also this line should be modified: https://github.com/kornia/kornia/blob/master/kornia/augmentation/augmentation3d.py#L944
> 
> plus, @shijianjian `RandomPerspective3D` don’t appear in the docs

Is this related with this pull request?

File: kornia/utils/grid.py [hunk 2]
```diff
@@ -57,8 +57,8 @@ def create_meshgrid(
         xs = (xs / (width - 1) - 0.5) * 2
         ys = (ys / (height - 1) - 0.5) * 2
     # generate grid by stacking coordinates
-    base_grid: torch.Tensor = torch.stack(torch.meshgrid([xs, ys])).transpose(1, 2)  # 2xHxW
-    return torch.unsqueeze(base_grid, dim=0).permute(0, 2, 3, 1)  # 1xHxWx2
+    base_grid: torch.Tensor = torch.stack(torch.meshgrid([ys, xs]), dim=-1)  # HxWx2
```
[the remark this answers is not in the document]
Comment (shiyangc-intusurg) [reply kornia/utils/grid.py:60]: In the original code, `torch.stack(torch.meshgrid([xs, ys]))` gives you 2xXxY, then `.transpose(1, 2)` gives you 2xYxX. This is equavalent of doing `torch.stack(torch.meshgrid([ys, xs]))`.
The reason I add `, dim=-1` is to remove the need of the `permute` in the next line.
Does this actually crash the test cases?

[the remark this answers is not in the document]
Comment (shiyangc-intusurg) [reply kornia/utils/grid.py]: Now the order of the coordinates is restored and the transpose and permute are merged together.

**Review by edgarriba — APPROVED** [review]
You might need to change the doctest from RandomPerspective3D

https://github.com/kornia/kornia/blob/master/kornia/augmentation/augmentation3d.py#L945
